src/mame/juku/juku.cpp

```diff
@@ -302,11 +337,60 @@ INPUT_PORTS_END
 //  VIDEO
 //**************************************************************************
 
+void juku_state::screen_cmd_a_w(uint8_t data) { m_pit[0]->write(0x11, data); screen_scan_sequence(0, data); }
+void juku_state::screen_data_a_w(uint8_t data) { m_pit[0]->write(0x12, data); screen_scan_sequence(1, data); }
+void juku_state::screen_cmd_b_w(uint8_t data) { m_pit[1]->write(0x15, data); m_screen_cur_seq[2] == -1 ? screen_scan_sequence(2, data) : screen_scan_sequence(3, data); }
+void juku_state::screen_data_b_w(uint8_t data) { m_pit[1]->write(0x16, data); screen_scan_sequence(4, data); }
+
+void juku_state::screen_scan_sequence(int pos, uint8_t data)
+{
+	m_screen_cur_seq[pos] = (int)data;
+
+	for (int i=0; i<3; i++)
+		if (memcmp(screen_mode_sequences[i], m_screen_cur_seq, pos*sizeof(int)) == 0) {
+			if (pos == 4)
+				switch (i) {
+				case 0:
+					screen_mode(320, 240);
+					break;
+				case 1:
+					screen_mode(384, 200);
+					break;
+				case 2:
+					screen_mode(400, 192);
+					break;
+				case 3:
+					screen_mode(256, 192);
+					break;
+				}
+			else return;
+		}
+
+	for (int i=0; i<5; i++)
+		m_screen_cur_seq[i]=-1;
+}
+
+void juku_state::screen_mode(int x, int y)
+{
+	if (m_screen_x == x && m_screen_y == y)
+		return;
+
+	m_screen_x = x, m_screen_y = y;
+	rectangle v = m_screen->visible_area();
+	v.max_x = x - 1;
+	v.max_y = y - 1;
+
+	m_screen->configure(x, y, v, m_screen->frame_period().attoseconds());
+}
```

Comment:
This doesn’t really seem that plausible.  Maintaining the same frame period for any combination of H/V counts isn’t something that’s achievable.  How is the pixel clock obtained?  Also, what kind of screen is this?  Unless it’s an LCD, it needs at least some vertical blanking lines – most CRT monitors require three lines of front porch, three lines of vertical sync, and at least a dozen lines of back porch.